lib/Common/Codex/Utf8Helper.h

```diff
@@ -142,14 +134,44 @@ namespace utf8
             // instead of replacing them with the "replacement" chracter. Pass a flag to our
             // decoder to require such behavior
             utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(remDestString, remSourceString, (LPCUTF8) sourceString + cbSourceString, DecodeOptions::doAllowInvalidWCHARs);
```

Comment:
The assert just before here that `cchDestString <= sourceCount` should be `cchDestString <= destBufferCount` I believe (maybe strictly less to allow for the null terminator), and if this function is exposed outside of our code then I think it should be an `if(...) { return E_OUTOFMEMORY;}` and not an assert.

Reply:
Not sure.. This PR didn't make any logical changes within the Utf8Helper stuff. Just split an existing interface for reuse purposes.

Reply:
Added `if (cchDestString > sourceCount)`